mesonbuild/msubprojects.py

```diff
@@ -444,7 +444,7 @@ def update(self) -> bool:
             success = True
         else:
             self.log('  -> Cannot update', self.wrap.type, 'subproject')
-        if success:
+        if success and os.path.exists(self.wrap_resolver.dirname):
```

Comment:
This is only an issue if we consider "cannot update subproject with no wrap file" to be success.

Hence it is part of the same commit IMO. But, I think we should check that more directly by skipping this in the case that we observed there is no wrap file. Maybe that should be the condition, even. Or maybe we should track this with `skipped = False`, and finally `return success or skipped`.

Reply:
No, it happens when updating all wraps and some have not been downloaded.

Reply:
update_file() returns success if dir does not exists.

Reply:
Changed to consider it success only when updating all subprojects.